### PR TITLE
SDCICD-607: Support creating and testing against STS clusters

### PR DESCRIPTION
**Pull request by jeefy**
Adds support for creating ROSA STS clusters.

Updates CIO and MVO tests to be skipped when it's an STS cluster.

Currently blocked by [SDA-4806](https://issues.redhat.com/browse/SDA-4806)
/hold

### CODE REVIEW

File: pkg/common/providers/ocmprovider/cluster.go [hunk 1]
```diff
@@ -294,6 +294,9 @@ func (o *OCMProvider) FindRecycledCluster(originalVersion, cloudProvider, produc
 				return ""
 			}
 			viper.Set(config.Cluster.Reused, true)
+			if recycledCluster.AWS().STS().RoleARN() != "" {
+				viper.Set("rosa.STS", true)
+			}
```
Comment (jharrington22) [inline pkg/common/providers/ocmprovider/cluster.go:299]: nit but can you do this just once above this block if its a recycled cluster?

**Comment by vkareh** [comment]
This LGTM

**Pull request merged**